spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -63,12 +68,33 @@ public JSONObject getQueryResponse(AsyncQueryJobMetadata asyncQueryJobMetadata)
     GetJobRunResult getJobRunResult =
         emrServerlessClient.getJobRunResult(
             asyncQueryJobMetadata.getApplicationId(), asyncQueryJobMetadata.getJobId());
-    JSONObject result = new JSONObject();
-    if (getJobRunResult.getJobRun().getState().equals(JobRunState.SUCCESS.toString())) {
-      result =
-          jobExecutionResponseReader.getResultFromOpensearchIndex(asyncQueryJobMetadata.getJobId());
+    String jobState = getJobRunResult.getJobRun().getState();
+    JSONObject result =
+        (jobState.equals(JobRunState.SUCCESS.toString()))
+            ? jobExecutionResponseReader.getResultFromOpensearchIndex(
+                asyncQueryJobMetadata.getJobId(), asyncQueryJobMetadata.getResultIndex())
+            : new JSONObject();
+
+    // if result index document has a status, we are gonna use the status directly; otherwise, we
+    // will use emr-s job status
+    // a job is successful does not mean there is no error in execution. For example, even if result
+    // index mapping
+    // is incorrect, we still write query result and let the job finish.
+    if (result.has(DATA_FIELD)) {
```

Comment:
We are setting result object only when job status from EMR is success. 

In case of index query, EMR JobRUn State is RUNNING, but the status should be succesful in the result.

Reply:
Probably we can merge and handle this use case as bug separately in a separate PR.

Reply:
let's track in issue https://github.com/opensearch-project/sql/issues/2214

Reply:
yeah, we can handle this use case a bug separately.